### PR TITLE
#3: Odp: Delete

### PR DESCRIPTION
**Pull request by sorja**
Part of #806

### CODE REVIEW

File: server/api/odp/deleteDraft.ts [hunk 1]
```diff
@@ -16,7 +15,8 @@ export const OdpDeleteDraft = {
           const { countryIso } = req.query
           await allowedToEditDataCheck(countryIso, req.user, 'extentOfForest')
 
-          const { odpId } = await db.transaction(OdpService.deleteDraft, [req.query.odpId, req.user])
+          // @ts-ignore
```
Comment (minotogna) [inline server/api/odp/deleteDraft.ts:18]: remove this. and cast to User

File: server/repository/odp/getAndCheckOdpCountryId.ts [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
+import { BaseProtocol, DB } from '@server/db'
+import { User } from '@core/auth'
 
-export const getAndCheckOdpCountryId = async (client: any, odpId: any, user: any) => {
-  const res = await client.query('SELECT country_iso FROM odp WHERE id = $1', [odpId])
-  const countryIso = res.rows[0].country_iso
+export const getAndCheckOdpCountryId = async (options: { odpId: number; user: User }, client: BaseProtocol = DB) => {
+  const { odpId, user } = options
+  const q = await client.query('SELECT country_iso FROM odp WHERE id = $1', [odpId])
+  // TODO: Remove after migration to pg-promise done
```
Comment (minotogna) [inline server/repository/odp/getAndCheckOdpCountryId.ts:8]: is this TODO still valid ?

Comment (sorja) [reply server/repository/odp/getAndCheckOdpCountryId.ts]: Currently yes, will be fixed in next PR

File: server/api/odp/delete.ts [hunk 3]
```diff
@@ -15,8 +14,9 @@ export const OdpDelete = {
         try {
           const { countryIso } = req.query
           await allowedToEditDataCheck(countryIso, req.user, 'extentOfForest')
-
-          await db.transaction(OdpService.deleteOdp, [req.query.odpId, req.user])
+          // @ts-ignore
+          // TODO: Add user type to express
+          await OdpService.deleteOdp({ odpId: Number(req.query.odpId), user: req.user })
```
Comment (minotogna) [inline server/api/odp/delete.ts:19]: remove this ts-ignore and todo. cast to User

**Review by minotogna — APPROVED** [review]

**Pull request merged**